packages/graphql/src/services/accounts.ts

```diff
@@ -150,69 +168,75 @@ export async function sendMagicLink(
   { usernameOrEmail }: SendMagicLinkArgs,
   ctx: Context
 ) {
-  const participantWithUsername = await ctx.prisma.participant.findUnique({
-    where: { username: usernameOrEmail.trim() },
-  })
-  const participantWithEmail = await ctx.prisma.participant.findUnique({
-    where: { email: usernameOrEmail.trim().toLowerCase() },
-  })
+  const trimmedUsernameOrEmail = usernameOrEmail.trim()
 
-  const participant = participantWithUsername || participantWithEmail
+  // the returned count can never be more than one, as the username cannot be a valid email (and vice versa)
+  const participantWithUsername = await ctx.prisma.participant.findMany({
+    where: {
+      OR: [
+        { username: trimmedUsernameOrEmail },
+        { email: trimmedUsernameOrEmail.toLowerCase() },
+      ],
+    },
+  })
 
-  // participant does not exist, return early
-  if (!participant) return true
+  if (participantWithUsername.length === 0) return true
 
-  const magicLinkToken = uuidv4()
-  const magicLinkExpiration = dayjs().add(15, 'minute').toDate()
+  const participantData = participantWithUsername[0]
 
-  // set loginToken and loginTokenExpiresAt
-  await ctx.prisma.participant.update({
-    where: { id: participant.id },
-    data: {
-      magicLinkToken: magicLinkToken,
-      magicLinkExpiresAt: magicLinkExpiration,
+  const magicLinkJWT = JWT.sign(
```

Comment:
using a JWT makes it independent of the db, we don't need to track expiration, and it is based on cryptography with our app secret (signed) -> still need to test that an outdated magic link will show some message ("has expired,try again...")